Panic in salvaging worker

```
6:31AM INF Salvaging a lost worker component=worker queueName=___masked___  queueUID=6cda697d-a712-481c-8856-81bfd412d3c6 salvageTargetUID=c0768ab7-cd16-4fe5-8bb5-8d5a530dc7c7 workerName=___masked___ workerUID=35ff1b4d-106e-49d4-adfd-83f66c2d0032

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x841245]
goroutine 1 [running]:
github.com/pfnet-research/pftaskqueue/pkg/backend/redis.(*Backend).SalvageWorker.func1.2(0xb6d0a0, 0xc00030e050, 0xc00030e050, 0x0)
        /workspace/pkg/backend/redis/worker.go:364 +0x755
github.com/go-redis/redis/v7.(*Pipeline).Pipelined(0xc00030e050, 0xc0000e83c0, 0xc00030e050, 0xa0, 0xa52cc0, 0xc0002fe101, 0xc0000e83c0)
        /go/pkg/mod/github.com/go-redis/redis/v7@v7.0.0-beta.5/pipeline.go:124 +0x43
github.com/go-redis/redis/v7.(*Tx).Pipelined(0xc0000f40e0, 0xc0000e83c0, 0x0, 0x0, 0x1, 0xc0002fe170, 0x0)
        /go/pkg/mod/github.com/go-redis/redis/v7@v7.0.0-beta.5/tx.go:140 +0x4c
github.com/go-redis/redis/v7.(*Tx).TxPipelined(...)
        /go/pkg/mod/github.com/go-redis/redis/v7@v7.0.0-beta.5/tx.go:145
github.com/pfnet-research/pftaskqueue/pkg/backend/redis.(*Backend).SalvageWorker.func1(0xc0000f40e0, 0xc000619750, 0x2)
        /workspace/pkg/backend/redis/worker.go:359 +0x11b2
github.com/go-redis/redis/v7.(*Client).WatchContext(0xc0002ec840, 0xb5ba00, 0xc0002fc900, 0xc0006199d8, 0xc000619750, 0x2, 0x2, 0x200, 0x9b0b40)
        /go/pkg/mod/github.com/go-redis/redis/v7@v7.0.0-beta.5/tx.go:83 +0x68
github.com/pfnet-research/pftaskqueue/pkg/backend/redis.(*Backend).runTxOnce(0xc0002d4a80, 0xb5ba00, 0xc0002fc900, 0xb56a00, 0xc0002d4000, 0x7, 0x0, 0x0, 0xc000576200, 0x1c4, ...)
        /workspace/pkg/backend/redis/redis.go:77 +0x81
github.com/pfnet-research/pftaskqueue/pkg/backend/redis.(*Backend).runTxWithBackOff.func1(0xc00056c0c0, 0xc00056c0c0)
        /workspace/pkg/backend/redis/redis.go:57 +0x337
github.com/cenkalti/backoff/v4.RetryNotifyWithTimer(0xc0006191a0, 0xb55b40, 0xc00056c0c0, 0xc000619190, 0xb599c0, 0xc0002fe000, 0x0, 0x0)
        /go/pkg/mod/github.com/cenkalti/backoff/v4@v4.0.0/retry.go:52 +0xf6
github.com/cenkalti/backoff/v4.RetryNotify(...)
        /go/pkg/mod/github.com/cenkalti/backoff/v4@v4.0.0/retry.go:31
github.com/pfnet-research/pftaskqueue/pkg/backend/redis.(*Backend).runTxWithBackOff(0xc0002d4a80, 0xb5ba00, 0xc0002fc900, 0xb56a00, 0xc0002d4000, 0x1, 0x0, 0x0, 0xc000576000, 0x1c4, ...)
        /workspace/pkg/backend/redis/redis.go:55 +0x2ae
github.com/pfnet-research/pftaskqueue/pkg/backend/redis.(*Backend).SalvageWorker(0xc0002d4a80, 0xb5ba00, 0xc0002fc900, 0x1c4812a77d69da6c, 0xc6d312d4bf815688, 0xd4496e104d1bff35, 0x32002d6cf683fdad, 0xe54f16cdb78a76c0, 0xc7c70d535a8db58b, 0x0, ...)
        /workspace/pkg/backend/redis/worker.go:383 +0x11ca
github.com/pfnet-research/pftaskqueue/pkg/worker.(*Worker).salvageLostWorkers(0xc0003c0000)
        /workspace/pkg/worker/worker.go:644 +0x39d
github.com/pfnet-research/pftaskqueue/pkg/worker.(*Worker).Start(0xc0003c0000, 0x0, 0x0)
        /workspace/pkg/worker/worker.go:116 +0xae
github.com/pfnet-research/pftaskqueue/cmd.glob..func24(0xf42460, 0xc000296820, 0x1, 0xd)
        /workspace/cmd/start_worker.go:76 +0x31
github.com/spf13/cobra.(*Command).execute(0xf42460, 0xc0002965b0, 0xd, 0xd, 0xf42460, 0xc0002965b0)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0xf41f60, 0x43b6aa, 0xf028c0, 0xc00008c750)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
github.com/pfnet-research/pftaskqueue/cmd.Execute()
        /workspace/cmd/root.go:130 +0x31
main.main()
        /workspace/main.go:25 +0x20
```

--- a/pkg/backend/redis/worker.go
+++ b/pkg/backend/redis/worker.go
@@ -360,7 +360,7 @@ func (b *Backend) SalvageWorker(ctx context.Context, queueUID, salvagingWorkerUI
 			pipe.Del(b.workerPendingTaskQueueKey(queue.UID.String(), salvageTargetWorker.UID.String()))
 			pipe.Set(b.workerKey(queue.UID.String(), salvageTargetWorker.UID.String()), marshaledSalvageTargetWorker, -1)
 			for i, t := range tasksToSalvage {
-				if len(t.Status.History) > 0 &&
+				if len(t.Status.History) > 0 && t.Status.History[len(t.Status.History)-1].SalvagedBy != nil &&
 					*t.Status.History[len(t.Status.History)-1].SalvagedBy == salvagingWorker.UID {
 					workerUID := t.Status.History[len(t.Status.History)-1].WorkerUID
 					pipe.SRem(b.workerTasksKey(queue.UID.String(), workerUID), t.UID)

